apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1729,18 +1735,35 @@ new_onchain_tx(channel_slash_tx, Opts,
     {ok, Tx} = new_onchain_tx_(aesc_slash_tx, Opts2, PinnedHeight),
     {ok, Tx, []}.
 
+-spec new_onchain_tx_(aesc_create_tx | aesc_withdraw_tx | aesc_deposit_tx |
+                      aesc_close_solo_tx | aesc_snapshot_solo_tx |
+                      aesc_slash_tx | aesc_settle_tx | aesc_close_mutual_tx,
+                      map(), non_neg_integer()) -> {ok, aetx:tx()}.
 new_onchain_tx_(Mod, Opts, CurrHeight) when Mod =:= aesc_create_tx;
                                             Mod =:= aesc_withdraw_tx;
                                             Mod =:= aesc_deposit_tx;
                                             Mod =:= aesc_close_solo_tx;
                                             Mod =:= aesc_snapshot_solo_tx;
                                             Mod =:= aesc_slash_tx;
-                                            Mod =:= aesc_settle_tx ->
-    case maps:is_key(fee, Opts) of
-        true -> %% use preset fee
+                                            Mod =:= aesc_settle_tx;
+                                            Mod =:= aesc_close_mutual_tx ->
+    FeeSpecified = maps:is_key(fee, Opts),
+    GasPriceSpecified = maps:is_key(gas_price, Opts),
+    case FeeSpecified of
+        true when not GasPriceSpecified -> %% use preset fee
             apply(Mod, new, [Opts]);
+        true when GasPriceSpecified -> %% both fee and gas_price are specified
+            {ok, Tx1} = apply(Mod, new, [Opts]),
+            {ok, Tx2} = create_with_minimum_fee(Mod, Opts#{fee => 0}, % gas_price is already in Opts
+                                                CurrHeight),
+            %% use the tx with the bigger fee
```

Comment:
Why the bigger? The user wants to pay extra?

Reply:
Yes, in the sense of _I want to pay at least that amount or at least that amount of `gas_price`_